File: frontend/src/app/product.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Product } from './product';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { environment } from 'src/environments/environment';
@Injectable({
  providedIn: 'root'
})
export class ProductService {

  constructor(private http: HttpClient) { }

  apiUrl = environment.apiUrl + "product";

  getProducts():Observable<Product[]> {
    return this.http.get<Product[]>(this.apiUrl);
  }

  getProduct(product:Product):Observable<Product> {
    return this.http.get<Product>(this.apiUrl + "/" + product.id);
  }

  createProduct(product : Product):Observable<Product> {
    return this.http.post<Product>(this.apiUrl , product);
  }

  deleteProduct(product : Product):Observable<Product> {
    return this.http.delete<Product>(this.apiUrl + "/" + product.id);
  }

  editProduct(product : Product):Observable<Product> {
    return this.http.put<Product>(this.apiUrl , product )
  }




}

File: product/src/main/java/com/iprobe/product/service/ProductService.java
package com.iprobe.product.service;

import com.iprobe.product.model.dto.ProductDTO;

import java.util.List;

public interface ProductService {
    List<ProductDTO> getProducts();
    ProductDTO getProduct(long id);
    ProductDTO createProduct(ProductDTO productDTO);
    ProductDTO updateProduct(ProductDTO productDTO);
    ProductDTO deleteProduct(long id);
}
